swag/parser.go

```diff
@@ -129,6 +129,13 @@ func NewParser(options ...func(*Parser)) *Parser {
 	return parser
 }
 
+// SetMarkdownFileDirectory sets the directory to search for markdownfiles
+func SetMarkdownFileDirectory(directoryPath string) func(*Parser) {
+	return func(p *Parser) {
+		p.markdownFileDir = directoryPath
+	}
```

Comment:
Why the need for indirection here? Why not just add a method, something like:

```
func (parser *Parser) SetMarkdownFileDirectory(directoryPath string) {
    parser.markdownFileDir = directoryPath
}
```

Reply:
This is actually the code from the current version of swag. So, my hope is we wouldn't need to touch anything if you update swag.